src/backends/torch/torchutils.cc

```diff
@@ -174,17 +186,27 @@ namespace dd
         {
           if (copied_params.find(param_name) == copied_params.end())
             {
-              logger->warn(param_name + " was not found in source module.");
+              if (logger)
+                {
+                  logger->warn(param_name
+                               + " was not found in source module.");
+                }
+
+              if (strict)
+                {
+                  throw MLLibBadParamException(
+                      param_name + " was not found in source module.");
+                }
             }
         }
     }
 
     void load_weights(torch::nn::Module &module, const std::string &filename,
                       const torch::Device &device,
-                      std::shared_ptr<spdlog::logger> logger)
+                      std::shared_ptr<spdlog::logger> logger, bool strict)
```

Comment:
`const bool &strict` ?

Reply:
references are pointers, and pointers are bigger than booleans, so unless there is a reason I don't see here, it is faster to pass booleans by value